web-gateway/src/main/java/controllers/AbstractController.java

```diff
@@ -47,17 +50,21 @@ public AbstractController(MessagesApi messagesApi, UserService userService) {
     }
 
     protected CompletionStage<Nav> loadNav(Optional<UUID> userId) {
-        return userService.getUsers(Optional.of(0), Optional.of(10)).invoke().thenApply(users -> {
-            Optional<User> currentUser = userId.flatMap(id -> {
-                for (User u: users.getItems()) {
-                    if (u.getId().equals(id)) {
-                        return Optional.of(u);
-                    }
+        CompletionStage<User> u = userService.getUser(userId.get())
+                .handleRequestHeader(authenticate(userId.get())).invoke();
```

Comment:
If `userId` is `empty` this will fail. Maybe invoke the remote service only if `userId.isPresent`. When not present there's no need to combine the `users` and `u` `CompletionStage`s.